docs/DeveloperGuide.adoc

```diff
@@ -192,6 +196,54 @@ image::add-ad.PNG[]
+
+[NOTE]
+If the user executes the `stats` command in the Todo mode, `StatsCommand` would throw a `CommandException` to the user specifying that the user is in the wrong mode. In addition, if there is no eateries with any reviews in the application at the time of `stats` execution, the `StatsCommand` would throw a `CommandException` error saying that there is no data to generate statistics. Lastly, if there is any error during the generating of statistics, the `StatsCommand` will throw another `CommandException` saying that there is an unknown error and the application is unable to generate statistics.
+
+. `StatsCommand` would then call `Model#setStatistics()` with `Statistics` as a a parameter.
+
+. After successfully generating the statistics, `StatsCommand` will return the `CommandResult` to the `Ui`.
+
+The following sequence diagram shows how the statistics feature is executed.
+image:StatsSequenceDiagram.png[]
+
+Since the statistic feature has a different user interface compared to the other commands i.e. `AddCommand`, `AddTagCommand`, etc., it would be useful to explain how the different user interface components work.
+
+. When the `Ui` receives the `CommandResult` from `StatsCommand`, the `Ui` will call `MainWindow#showStats()`.
+
+. `MainWindow#showStats()` will then if `StatsWindow` is already open, if yes then it will call `StatsWindow#initStats()` and focuses the window. Else, it will call `StatsWindow#initStats()` and shows the window.
+
+. In `StatsWindow#initStats()`, `StatsWindow` will create a `CustomLayeredGraph` and `CustomPiechart` to create the graph and pie chart needed for the statistics report.
+
+The following sequence diagram shows how the statistics feature shows the statistics to the user.
+image:StatsUiSequenceDiagram.png[]
+
+==== Design Considerations
+===== Aspect: Whether the statistics should auto-update an eatery is added or a review is added to an eatery.
+* ** Alternative 1 (current choice):** Not making the statistics auto-update
+*** Pros: Easy to implement; don't need to refactor existing code to make use of ObservableList and add ListChange listeners.
+*** Cons: User would need to call `stats` command after every addition or change made to the eatery or reviews of an eatery.
+
+* ** Alternative 2: ** Making the statistics auto-update
+*** Pros: User would not need to call `stats` command after every addition or change made to the eatery or reviews of an eatery.
+*** Cons: Need to refactor a lot of existing code.
+
+Reason for choosing alternative 1: Although it may be annoying for the user to keep calling `stats` command after every change made, we realise that since `stats` acts as a statistics report for the users, users may not call the `stats` command as often to justify refactoring the entire code.
```

Comment:
actually another reason u can use also is that you stats appear in a separate window, so for them to edit their reviews, they would either [x] it or minimise, so in this case by making them x it and reopening does not change the steps they need to see a change in their stats. Ie, even if its autoupdated, they still have to minimise and unminimise the window.